cl/recap/tasks.py

```diff
@@ -823,6 +829,97 @@ def process_recap_docket_history_report(self, pk):
     }
 
 
+@app.task(
+    bind=True, max_retries=3, interval_start=5 * 60, interval_step=5 * 60
+)
+def process_case_query_page(self, pk):
+    """Process the case query (iquery.pl) page.
+
+    :param pk: The primary key of the processing queue item you want to work on
+    :returns: A dict indicating whether the docket needs Solr re-indexing.
+    """
+
+    pq = ProcessingQueue.objects.get(pk=pk)
+    mark_pq_status(pq, "", PROCESSING_STATUS.IN_PROGRESS)
+    logger.info(f"Processing RECAP item (debug is: {pq.debug}): {pq}")
+
+    try:
+        text = pq.filepath_local.read().decode()
+    except IOError as exc:
+        msg = f"Internal processing error ({exc.errno}: {exc.strerror})."
+        if (self.request.retries == self.max_retries) or pq.debug:
+            mark_pq_status(pq, msg, PROCESSING_STATUS.FAILED)
+            return None
+        else:
+            mark_pq_status(pq, msg, PROCESSING_STATUS.QUEUED_FOR_RETRY)
+            raise self.retry(exc=exc)
+
+    report = CaseQuery(map_cl_to_pacer_id(pq.court_id))
+    report._parse_text(text)
+    data = report.data
+    logger.info(f"Parsing completed for item {pq}")
+
+    if data == {}:
+        # Bad docket iquery page.
+        msg = "Not a valid case query page upload."
+        mark_pq_status(pq, msg, PROCESSING_STATUS.INVALID_CONTENT)
+        self.request.chain = None
+        return None
+
+    # Merge the contents of the docket into CL.
+    d = find_docket_object(
+        pq.court_id, pq.pacer_case_id, data["docket_number"]
+    )
+    content_updated = False
+    current_case_name = d.case_name
+    d.add_recap_source()
+    update_docket_metadata(d, data)
+
+    if current_case_name != d.case_name or not d.pk:
+        # This docket should be added to Solr or updated since is new or the
+        # case name has changed.
+        content_updated = True
```

Comment:
Two things. The search engine only works if we have RECAPDocument items in a docket. Without a RECAPDocument object, we can't add it to Solr, unfortunately. All case query pages do not have enough information to make a RECAP Document for the docket, so if this is a new docket, that's great, BUT no reason to add it to solr (it lacks any RECAP Documents).

Second, we should put `content_updated` closer to where it's doing work.

So:

```suggestion
    current_case_name = d.case_name
    d.add_recap_source()
    update_docket_metadata(d, data)
    
    content_updated = False
    if current_case_name != d.case_name:
        # Update the docket in PACER since is new or the
        # case name has changed.
        content_updated = True
```